src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -70,14 +70,6 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
-            // TODO: Investigate why this check is needed.
-            if (Propagators.DefaultTextMapPropagator.Extract(default, request, HttpRequestMessageContextPropagation.HeaderValuesGetter) != default)
```

Comment:
@denisivan0v @cijothomas 

FYI I took this for a spin too. HttpClient looks much better!

![image](https://user-images.githubusercontent.com/28367120/174141539-45506622-2d9e-4b81-a47c-de47ce613142.png)

@denisivan0v Would you do me a favor and add this test to your PR?

* Modify the server listener like this...

```csharp
        public HttpClientTests()
        {
            this.serverLifeTime = TestHttpServer.RunServer(
                (ctx) =>
                {
                    if (ctx.Request.Url.PathAndQuery.Contains("redirect"))
                    {
                        ctx.Response.RedirectLocation = "/";
                        ctx.Response.StatusCode = 302;
                    }
                    else
                    {
                        ctx.Response.StatusCode = 200;
                    }

                    ctx.Response.OutputStream.Close();
                },
                out var host,
                out var port);

            this.url = $"http://{host}:{port}/";
        }
```

* Add this test...

```csharp
        [Fact]
        public async Task HttpClientRedirectTest()
        {
            var processor = new Mock<BaseProcessor<Activity>>();
            using (Sdk.CreateTracerProviderBuilder()
                .AddHttpClientInstrumentation()
                .AddProcessor(processor.Object)
                .Build())
            {
                using var c = new HttpClient();
                await c.GetAsync($"{this.url}redirect");
            }

            Assert.Equal(7, processor.Invocations.Count); // SetParentProvider/OnShutdown/Dispose/OnStart/OnEnd/OnStart/OnEnd called.

            var firstActivity = (Activity)processor.Invocations[2].Arguments[0]; // First OnEnd

            Assert.Contains(firstActivity.TagObjects, t => t.Key == "http.status_code" && (int)t.Value == 302);

            var secondActivity = (Activity)processor.Invocations[4].Arguments[0]; // Second OnEnd

            Assert.Contains(secondActivity.TagObjects, t => t.Key == "http.status_code" && (int)t.Value == 200);
        }
```

That verifies for automatic redirect 2 spans are created and the status codes match.

Reply:
@denisivan0v Could you add this test for redirect specifically? OR is this already covered by existing tests?

Reply:
@cijothomas, @CodeBlanch added requested changes and the test for redirect, thanks! And sorry for missing this earlier :)